apps/design/backend/src/candidate_rotation.ts

```diff
@@ -0,0 +1,65 @@
+import { assertDefined, iter } from '@votingworks/basics';
+import { AnyContest } from '@votingworks/types';
+
+// Maps the number of candidates in a contest to the index at which to rotate
+// the candidates. These indexes are randomly selected by the state every 2
+// years. Note that these use 1-based indexing.
+const NH_ROTATION_INDICES: Record<number, number> = {
+  2: 2,
+  3: 1,
+  4: 2,
+  5: 1,
+  6: 1,
+  7: 7,
+  8: 7,
+  9: 2,
+  10: 6,
+  11: 3,
+  12: 3,
+  13: 8,
+  14: 11,
+  15: 4,
+  16: 5,
+  17: 7,
+  18: 18,
+  19: 19,
+  20: 18,
+};
+
+/**
+ * Rotate a contest's candidates according to the governing statutes. Currently
+ * only supports NH rules.
+ *
+ * The NH rotation algorithm is as follows:
+ * 1. Order the candidates alphabetically by last name.
+ * 2. Cut the "deck" at a randomly selected index (see NH_ROTATION_INDICES).
+ */
+export function rotateCandidates(contest: AnyContest): AnyContest {
```

Comment:
Slightly nicer type would avoid casting in other places:

```suggestion
export function rotateCandidates<C extends AnyContest>(contest: C): C {
```

Reply:
I like it, but it made ESLint complain :/

<img width="869" alt="Screenshot 2024-01-16 at 1 59 42 PM" src="https://github.com/votingworks/vxsuite/assets/530106/c0abd7bd-bf5f-45fc-a411-62a2e057037e">